slimeworldmanager-plugin/src/main/java/com/grinderwolf/swm/plugin/commands/sub/DeleteWorldCmd.java

```diff
@@ -138,5 +141,33 @@ public boolean onCommand(CommandSender sender, String[] args) {
 
         return false;
     }
+
+    @Override
+    public List<String> onTabComplete(CommandSender sender, String[] args) {
+        if (args.length == 2) {
+            final List<String> toReturn = new LinkedList<>();
```

Comment:
I'd change this to the following:

```java
            final List<String> toReturn = null;
            final String typed = args[1].toLowerCase();

            for (World world : Bukkit.getWorlds()) {
                final String worldName = world.getName();

                if (worldName.toLowerCase().startsWith(typed)) {
                    if (toReturn == null) {
                        toReturn = new LinkedList<>();
                    }
                    
                    toReturn.add(worldName);
                }
            }

            return toReturn == null ? Collections.emptyList() : toReturn;
```

By doing this, if no worlds are found, you won't be creating an empty list every time.

Reply:
Already thought of that, will add this! :) Thanks!